--- /dev/null
+++ ../lib/PhpParser/Node/Stmt/Do_.php
@@ -4,7 +4,7 @@
 
 use PhpParser\Node;
 
-class Do_ extends Node\Stmt
+class Do_ extends Node\Stmt implements \Rector\Contract\PhpParser\Node\StmtsAwareInterface
 {
     /** @var Node\Stmt[] Statements */
     public $stmts;
@@ -27,7 +27,7 @@
     public function getSubNodeNames() : array {
         return ['stmts', 'cond'];
     }
-    
+
     public function getType() : string {
         return 'Stmt_Do';
     }
