--- /dev/null
+++ ../lib/PhpParser/Node/Stmt/For_.php
@@ -4,7 +4,7 @@
 
 use PhpParser\Node;
 
-class For_ extends Node\Stmt
+class For_ extends Node\Stmt implements \Rector\Contract\PhpParser\Node\StmtsAwareInterface
 {
     /** @var Node\Expr[] Init expressions */
     public $init;
@@ -36,7 +36,7 @@
     public function getSubNodeNames() : array {
         return ['init', 'cond', 'loop', 'stmts'];
     }
-    
+
     public function getType() : string {
         return 'Stmt_For';
     }
