--- /dev/null
+++ ../lib/PhpParser/Node/Stmt/Foreach_.php
@@ -4,7 +4,7 @@
 
 use PhpParser\Node;
 
-class Foreach_ extends Node\Stmt
+class Foreach_ extends Node\Stmt implements \Rector\Contract\PhpParser\Node\StmtsAwareInterface
 {
     /** @var Node\Expr Expression to iterate */
     public $expr;
@@ -40,7 +40,7 @@
     public function getSubNodeNames() : array {
         return ['expr', 'keyVar', 'byRef', 'valueVar', 'stmts'];
     }
-    
+
     public function getType() : string {
         return 'Stmt_Foreach';
     }
