--- /dev/null
+++ ../lib/PhpParser/Node/Stmt/If_.php
@@ -4,7 +4,7 @@
 
 use PhpParser\Node;
 
-class If_ extends Node\Stmt
+class If_ extends Node\Stmt implements \Rector\Contract\PhpParser\Node\StmtsAwareInterface
 {
     /** @var Node\Expr Condition expression */
     public $cond;
@@ -36,7 +36,7 @@
     public function getSubNodeNames() : array {
         return ['cond', 'stmts', 'elseifs', 'else'];
     }
-    
+
     public function getType() : string {
         return 'Stmt_If';
     }
