--- /dev/null
+++ ../Loader/PhpFileLoader.php
@@ -11,6 +11,7 @@

 namespace Symfony\Component\DependencyInjection\Loader;

+use Rector\Config\RectorConfig;
 use Symfony\Component\Config\Builder\ConfigBuilderGenerator;
 use Symfony\Component\Config\Builder\ConfigBuilderGeneratorInterface;
 use Symfony\Component\Config\Builder\ConfigBuilderInterface;
@@ -64,7 +65,7 @@
             $callback = $load($path, $this->env);

             if (\is_object($callback) && \is_callable($callback)) {
-                $this->executeCallback($callback, new ContainerConfigurator($this->container, $this, $this->instanceof, $path, $resource, $this->env), $path);
+                $this->executeCallback($callback, new RectorConfig($this->container, $this, $this->instanceof, $path, $resource, $this->env), $path);
             }
         } finally {
             $this->instanceof = [];
@@ -121,6 +122,9 @@
             }
             $type = $reflectionType->getName();
             switch ($type) {
+                case RectorConfig::class:
+                    $arguments[] = $containerConfigurator;
+                    break;
                 case ContainerConfigurator::class:
                     $arguments[] = $containerConfigurator;
                     break;
