--- /dev/null
+++ ../Helper/Helper.php
@@ -46,6 +46,7 @@
     public static function width(?string $string): int
     {
         $string ??= '';
+        // return early, to prevent a dependency on symfony/string to ease downgrading rector.
+        return \mb_strwidth($string);

         if (preg_match('//u', $string)) {
             return (new UnicodeString($string))->width(false);
@@ -65,6 +66,7 @@
     public static function length(?string $string): int
     {
         $string ??= '';
+        // return early, to prevent a dependency on symfony/string to ease downgrading rector.
+        return \mb_strlen($string);

         if (preg_match('//u', $string)) {
             return (new UnicodeString($string))->length();
